manager/job/sync_peers.go

```diff
@@ -175,84 +227,87 @@ func (s *syncPeers) createSyncPeers(ctx context.Context, scheduler models.Schedu
 
 // Merge sync peer results with the data in the peer table.
 func (s *syncPeers) mergePeers(ctx context.Context, scheduler models.Scheduler, results []*resource.Host, log *logger.SugaredLoggerOnWith) {
-	// Convert sync peer results from slice to map.
-	syncPeers := make(map[string]*resource.Host)
-	for _, result := range results {
-		syncPeers[result.ID] = result
-	}
-
-	rows, err := s.db.Model(&models.Peer{}).Where("scheduler_cluster_id = ?", scheduler.SchedulerClusterID).Rows()
-	if err != nil {
+	// Fetch existing peers from the database
+	var existingPeers []models.Peer
+	if err := s.db.Model(&models.Peer{}).Where("scheduler_cluster_id = ?", scheduler.SchedulerClusterID).Find(&existingPeers).Error; err != nil {
```

Comment:
Query by `models.Paginate`, refer to `https://github.com/dragonflyoss/Dragonfly2/blob/main/manager/service/peer.go#L80`.